---
layout: post
num: 2
title: Exercise 2
chapter: Chapter 12
description: >
  (1) Define an object reference and initialize it to null. Try to call a method through this reference. Now wrap the code in a try-catch clause to catch the exception.
---



```java


package com.ciaoshen.thinkinjava.chapter12;

public class Exercise2 {
    public void foo() {System.out.println("This is Exercise 2 of chapter 12");}
    public static void main(String[] args) {
        try {
            Exercise2 myExercise = null;
            myExercise.foo();
        } catch(Exception e) {
            System.out.println(e);
        }
    }
}


```



